build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/distribution/ElasticsearchDistributionExtension.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.distribution;
+
+import org.elasticsearch.gradle.plugin.PluginPropertiesExtension;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.tasks.AbstractCopyTask;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.io.File;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import static org.elasticsearch.gradle.plugin.PluginBuildPlugin.EXPLODED_BUNDLE_CONFIG;
+
+public class ElasticsearchDistributionExtension {
+    public static final String CONFIG_BIN_REGEX = "([^\\/]+\\/)?(config|bin)\\/.*";
+    private final Project project;
+
+    public ElasticsearchDistributionExtension(Project project) {
+        this.project = project;
+    }
+
+    private Configuration moduleZip(Project module) {
+        Map<String, String> moduleConfigurationCoords = Map.of("path", module.getPath(), "configuration", EXPLODED_BUNDLE_CONFIG);
+        Dependency dep = project.getDependencies().project(moduleConfigurationCoords);
+        return project.getConfigurations().detachedConfiguration(dep);
+    }
+
+    public void copyModule(TaskProvider<AbstractCopyTask> copyTask, Project module) {
```

Comment:
Move these methods into a plugin to make them a) plain java and more efficient and b) better testable. We'll likely move more of the logic from :distribution into testable plugin code over time